test/lib/server/services/qualityControlFlag/QcFlagService.test.js

```diff
@@ -230,4 +230,352 @@ module.exports = () => {
             expect(fetchedSortedProperties).to.have.all.ordered.members([...fetchedSortedProperties].sort().reverse());
         });
     });
+
+    describe('Creating Quality Control Flag for data pass', () => {
+        it('should fail to create quality control flag due to incorrect external user id', async () => {
+            const qcFlagCreationParameters = {
+                from: new Date('2019-08-09 01:29:50').getTime(),
+                to: new Date('2019-08-09 05:40:00').getTime(),
+                comment: 'VERY INTERESTING REMARK',
+            };
+
+            const relations = {
+                user: {
+                    externalUserId: 9999999, // Failing property
+                },
+                flagTypeId: 2,
+                runNumber: 106,
+                dataPassId: 1,
+                dplDetectorId: 1,
+            };
+
+            await assert.rejects(
+                () => qcFlagService.createForDataPass(qcFlagCreationParameters, relations),
+                new BadParameterError('User with this external id (9999999) could not be found'),
+            );
+        });
+
+        it('should fail to create quality control flag because qc flag `from` timestamp is smaller than run.startTime', async () => {
+            const qcFlagCreationParameters = {
+                from: new Date('2019-08-08 11:36:40').getTime(), // Failing property
+                to: new Date('2019-08-09 05:40:00').getTime(),
+                comment: 'VERY INTERESTING REMARK',
+            };
+
+            const relations = {
+                user: {
+                    externalUserId: 456,
+                },
+                flagTypeId: 2,
+                runNumber: 106,
+                dataPassId: 1,
+                dplDetectorId: 1,
+            };
+
+            await assert.rejects(
+                () => qcFlagService.createForDataPass(qcFlagCreationParameters, relations),
+                // eslint-disable-next-line max-len
+                new BadParameterError(`Given QC flag period (${new Date('2019-08-08 11:36:40').getTime()}, ${new Date('2019-08-09 05:40:00').getTime()}) is out of run (${new Date('2019-08-08 13:00:00').getTime()}, ${new Date('2019-08-09 14:00:00').getTime()}) period`),
+            );
+        });
+
+        it('should fail to create quality control flag because qc flag `from` timestamp is `to` timestamp', async () => {
```

Comment:
```suggestion
        it('should fail to create quality control flag because qc flag `from` timestamp is greater than `to` timestamp', async () => {
```

Reply:
updated